stable_baselines/common/policies.py

```diff
@@ -181,36 +217,81 @@ class ActorCriticPolicy(BasePolicy):
-            self.deterministic_action = self.proba_distribution.mode()
-            self.neglogp = self.proba_distribution.neglogp(self.action)
+            self._action = self.proba_distribution.sample()
+            self._deterministic_action = self.proba_distribution.mode()
+            self._neglogp = self.proba_distribution.neglogp(self.action)
             if isinstance(self.proba_distribution, CategoricalProbabilityDistribution):
-                self.policy_proba = tf.nn.softmax(self.policy)
+                self._policy_proba = tf.nn.softmax(self.policy)
             elif isinstance(self.proba_distribution, DiagGaussianProbabilityDistribution):
-                self.policy_proba = [self.proba_distribution.mean, self.proba_distribution.std]
+                self._policy_proba = [self.proba_distribution.mean, self.proba_distribution.std]
             elif isinstance(self.proba_distribution, BernoulliProbabilityDistribution):
-                self.policy_proba = tf.nn.sigmoid(self.policy)
+                self._policy_proba = tf.nn.sigmoid(self.policy)
             elif isinstance(self.proba_distribution, MultiCategoricalProbabilityDistribution):
-                self.policy_proba = [tf.nn.softmax(categorical.flatparam())
+                self._policy_proba = [tf.nn.softmax(categorical.flatparam())
                                      for categorical in self.proba_distribution.categoricals]
             else:
-                self.policy_proba = []  # it will return nothing, as it is not implemented
-            self._value = self.value_fn[:, 0]
-
+                self._policy_proba = []  # it will return nothing, as it is not implemented
+            self._value_tensor = self.value_fn[:, 0]
+
+    @property
+    def pdtype(self):
+        """ProbabilityDistributionType: type of the distribution for stochastic actions."""
+        return self._pdtype
+
+    @property
+    def policy(self):
+        """tf.Tensor: policy output, e.g. logits."""
+        return self._policy
+
+    @property
+    def proba_distribution(self):
+        """ProbabilityDistribution: distribution of stochastic actions."""
+        return self._proba_distribution
+
+    @property
+    def value_fn(self):
+        """tf.Tensor: value estimate, of shape (self.n_batch, 1)"""
+        return self._value_fn
+
+    @property
+    def value_flat(self):
+        """tf.Tensor: value estimate, of shape (self.n_batch, )"""
+        return self._value_tensor
+
+    @property
+    def action(self):
+        """tf.Tensor: stochastic action, of shape (self.n_batch, ) + self.ac_space.shape."""
+        return self._action
+
+    @property
+    def deterministic_action(self):
+        """tf.Tensor: deterministic action, of shape (self.n_batch, ) + self.ac_space.shape."""
+        return self._deterministic_action
+
+    @property
+    def neglogp(self):
+        """tf.Tensor: negative log probability of the action sampled by self.action."""
```

Comment:
```suggestion
        """tf.Tensor: negative log likelihood of the action sampled by self.action."""
```
this is to avoid talking about the probability of taking an action in the continuous case (which is exactly zero)